src/components/modal/Modal.svelte

```diff
@@ -0,0 +1,408 @@
+  // Whether show mask or not
+  export let mask = true;
+  // Close the modal when the mask is clicked
+  export let maskClosable = true;
+  // Function to be called after the modal closes
+  export let afterClose = () => {};
+  // Functions for destroying the modal
+  export const destroy = () => {
+    localVisible = false;
+    scrollableBody = true;
+    afterClose();
+    // Remove the destroy function modalDestroyFunctions store
+    if (modalType !== "standard") {
+      modalDestroyFunctions.set(
+        $modalDestroyFunctions.filter(func => func !== destroy)
+      );
+    }
+  };
+  // Body style for modal body element
+  export let bodyStyle = "";
+  // Style for the mask
+  export let maskStyle = "";
+  // The classname for the wrapper
+  export let wrapClassName = "";
+  // Custom z-index
+  export let zIndex = 1000;
+  // Focuses on the button when the modal opens
+  export let autoFocusButton = "ok";
+  // Allow user to close the modal if ESC is pressed
+  export let keyboard = true;
+
+  // ********************** /Props **********************
+
+  // Need to allow clickPosition to be set before first transition
+  let localVisible = false;
+  // Used so we can set a transform-origin for the transition
+  let clickPosition = null;
+  // Used for custom transition.
+  let fadescale = function(node, { duration }) {
+    return {
+      duration,
+      css: t => `opacity: ${t}; transform: scale(${t});`
+    };
+  };
+  // Body should not be scrollable when the modal is open
+  let scrollableBody = true;
+  // Confirm and status modals need special classes
+  let modalClasses;
+  // Style string for the ant-modal element
+  let modalStyleString = width ? `width: ${width};` : "";
+  // Ref for the root element
+  let thisModal;
+
+  function setClickPosition(e) {
+    clickPosition = {
+      x: e.pageX,
+      y: e.pageY
+    };
+  }
+
+  function setWrapClickClose(e) {
+    if (e.target.classList.value.includes("ant-modal-wrap")) {
+      onCancelLocal();
+    }
+  }
+
+  function removeModalOnEscape(e) {
+    if (e.key === "Escape" || e.keyCode === 27) {
+      const allModalsOpen = document.querySelectorAll(".ant-modal-root");
+      const lastModal = allModalsOpen[allModalsOpen.length - 1];
+      if (thisModal === lastModal) {
+        onCancelLocal();
+      }
+    }
+  }
+
+  onMount(() => {
+    if (keyboard) {
+      window.addEventListener("keydown", removeModalOnEscape);
+    }
+
+    // Add destroy function to the store so we can call Modal.destroyAll()
+    if (modalType !== "standard") {
+      modalDestroyFunctions.set([...$modalDestroyFunctions, destroy]);
+    }
+  });
+
+  onDestroy(() => {
+    if (typeof window !== "undefined") {
+      window.removeEventListener("keydown", removeModalOnEscape);
+      document.body.removeEventListener("click", setWrapClickClose);
+      document.removeEventListener("click", setClickPosition);
+    }
+  });
+
+  $: modalClasses = computeModalClasses();
```

Comment:
I think we could use the `classnames` package for this.
example: https://github.com/svelting/svant/blob/master/src/components/button/Button.svelte#L124